### PR TITLE
Fix Style/LineLength

### PR DESCRIPTION
**Pull request by macanudo527**
#### Second Batch of Layout/LineLength Rubocop Offense Fixes

- Helps #6055 

This makes code easier to read on smaller screens by reducing the line length.

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/models/product_importer_spec.rb
Comment (mkllnk) [inline spec/models/product_importer_spec.rb:547]: This is a lot harder to read. Maybe we should define the CSV as plain text, possibly with HEREDOC?

**Review by mkllnk — COMMENTED** [review]
I made it! :tada: 

193 files later... The code looks so much better now. There are only very few changes which don't improve readability in my eyes. I would have appreciated smaller chunks. And I like to separate auto-correct from manual changes. One way is to let it autocorrect and then only commit the changes we are sure about. Then the manual fixes can be addressed separately.

We also prefer rebase on master over merge. I just realised that we don't have to review merge commits when rebase is used.

Don't get the wrong impression from my nitpicking though. This is a great contribution.

**Comment by macanudo527** [comment]
> We also prefer rebase on master over merge. I just realised that we don't have to review merge commits when rebase is used.

Yeah, I was trying to do that, but was just going around in circles. I had a hard time rebasing with so many commits and files. Obviously I learned my lesson to keep PRs short and simple, and then rebasing will be a lot easier.

File: app/models/spree/app_configuration.rb
Comment (macanudo527) [reply app/models/spree/app_configuration.rb:79]: It seems like the settings in `image.rb` are used? 

[https://github.com/openfoodfoundation/openfoodnetwork/issues/5911#issuecomment-721162756](https://github.com/openfoodfoundation/openfoodnetwork/issues/5911#issuecomment-721162756)

File: app/controllers/api/v0/order_cycles_controller.rb
Comment (macanudo527) [reply app/controllers/api/v0/order_cycles_controller.rb:83]: I guess I've never dealt with `ransack` before. Something like this should be okay, right?
```
def permitted_ransack_params
  [:name OR :meta_keywords OR :variants_display_as OR
   :variants_display_name OR :supplier_name_cont,
   :with_properties, :primary_taxon_id_in_any]
end
```

File: app/helpers/order_cycles_helper.rb
Comment (macanudo527) [reply app/helpers/order_cycles_helper.rb:78]: You mean after `&&`? I don't think it is valid to start a line with `&&` or at least I keep getting `syntax error` when I do.

File: app/models/spree/app_configuration.rb
Comment (mkllnk) [reply app/models/spree/app_configuration.rb:79]: Yes, I just searched with `git grep attachment_styles` and nothing came up except this line here. So this can be removed, in a separate commit.

File: app/controllers/api/v0/order_cycles_controller.rb
Comment (mkllnk) [reply app/controllers/api/v0/order_cycles_controller.rb:83]: Yes, there's one JS example in the code:
https://github.com/openfoodfoundation/openfoodnetwork/blob/7cd1b5d5db69d24970a9a6d93096237895338bbf/app/assets/javascripts/admin/line_items/controllers/line_items_controller.js.coffee#L21-L28

File: app/helpers/order_cycles_helper.rb
Comment (mkllnk) [reply app/helpers/order_cycles_helper.rb:78]: Oh sorry, I didn't know that the line couldn't start with `&&`. After is fine then.

**Comment by macanudo527** [comment]
Sorry about that I force pushed before the previous tests finished and it looks like the database didn't get reset properly. You'll have to rerun the tests.

But, on the bright side I learned how to properly rebase. (I think)

File: app/models/spree/preferences/store.rb
Comment (mkllnk) [inline app/models/spree/preferences/store.rb:42]: This could blow up on certain commands. You're trying to read from the table before checking if it exists.

File: app/controllers/api/v0/order_cycles_controller.rb
Comment (mkllnk) [inline app/controllers/api/v0/order_cycles_controller.rb:86]: Does this work? I thought it needed to be joined with `_or_`.

File: app/models/spree/order.rb
Comment (mkllnk) [inline app/models/spree/order.rb:145]: A multi-line string is fine here, possibly nicer to read. No need for `squish` but it doesn't matter.

File: spec/models/product_importer_spec.rb
Comment (mkllnk) [inline spec/models/product_importer_spec.rb:543]: What's the difference between `csv_doc` and `csv_data`? Just wondering if we need this line. It doesn't hurt though.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Well done. All good except one condition that could raise an error now.

File: app/controllers/api/v0/order_cycles_controller.rb
Comment (macanudo527) [reply app/controllers/api/v0/order_cycles_controller.rb:86]: The spec passes with either, so I'll just update it to `_or_` to be safe.

File: app/models/spree/order.rb
Comment (macanudo527) [reply app/models/spree/order.rb:145]: It's a style thing. Without the `squish`, the `SQL` on line 145 has to have no indentation, so it would appear all the way to left. It was bugging me.

In the future, should I leave it un-squished?

File: spec/models/product_importer_spec.rb
Comment (macanudo527) [reply spec/models/product_importer_spec.rb:543]: For some strange reason I thought I needed to generate it. It works without it though so I removed it.

File: app/models/spree/order.rb
Comment (mkllnk) [reply app/models/spree/order.rb:145]: It doesn't have to be on the left, you can have indentation in SQL statements. Anyway, it doesn't matter.

File: app/models/spree/preferences/store.rb
Comment (mkllnk) [inline app/models/spree/preferences/store.rb:43]: Much better.

File: app/controllers/api/v0/order_cycles_controller.rb
Comment (mkllnk) [reply app/controllers/api/v0/order_cycles_controller.rb:86]: I don't see that change. Is this covered by the specs? What happens if you intentionally add a typo? Does it still pass?

Comment (macanudo527) [reply app/controllers/api/v0/order_cycles_controller.rb:86]: It passes with an intentional typo. Should we write a spec for it?

Comment (mkllnk) [reply app/controllers/api/v0/order_cycles_controller.rb:86]: Yes. Good idea.

Comment (macanudo527) [reply app/controllers/api/v0/order_cycles_controller.rb]: Sorry I've been away for a month. I was trying to figure out how to write a spec for this and I'm stumped. The best I could do is just check the parameter against itself, which doesn't seem very robust.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Good work. You are experiencing the downside of creating a big pull request that can then hang around for a while. Smaller chunks usually get through review and testing quicker and then the conflict resolution is often not needed or it's easier to do.

There's a failing test case as well.

You have two options:

* Try to fix this PR including everything.
* Reduce this PR to only the good parts that don't have any issues and get it reviewed and merged while you work on a solution for the other parts.

**Comment by macanudo527** [comment]
> Good work. You are experiencing the downside of creating a big pull request that can then hang around for a while. Smaller chunks usually get through review and testing quicker and then the conflict resolution is often not needed or it's easier to do.

I've definitely learned my lesson with this one. Short and simple is easy to review and easy to merge.

I'll try to pare this one down and just get it passed.

File: spec/controllers/api/v0/order_cycles_controller_spec.rb
Comment (macanudo527) [reply spec/controllers/api/v0/order_cycles_controller_spec.rb:40]: That makes more sense.

**Comment by macanudo527** [comment]
I'm very confused at this point. A lot of the tests that have failed are failing on my master branch as well. Am I missing something? It seems like I get a lot of failed tests even when I try to restore files I edited. 

In general, I get a lot of failed tests whenever I submit a PR for files I haven't even touched. Is this normal? I guess this is my first time working with a project with this many tests, but it seems like code even in the master branch fails tests at least for me locally.

I've  tried going with the docker setup locally, and capybara tests don't work, so that is not very helpful. I tried working without docker and it seems to work better sometimes. I really do want to contribute, but the setup is incredibly difficult. I might just have to abandon this PR and project.

**Comment by mkllnk** [comment]
Hey Neal, I'm sorry to hear that. The Docker setup is still experimental and we haven't solved the problem with system specs yet. There seems to be an issue connecting to Chrome but we don't know why. All core contributors install OFN on their host system without Docker.

Some specs seem to fail related to `variant_unit` which just underwent a huge refactor. So I assume some merge conflicts caused this.

I often use `git bisect` to find which commit is causing issues. Another approach is to push to your branch commit by commit. So if you push only one commit first and see if the specs pass then you can push the next one and see when it breaks.

**Comment by jibees** [comment]
Hi @macanudo527 

Sorry to hear that. Would like to help!
Some points:
 - The docker setup is unfortunately poorly updated since we don't have any core developper using it. It's a shame I know, but we miss time ;)
 - As already said, having a PR with a lot of modified files can be a pain (there will be more and more conflicts). 
 - Getting constantly failing tests locally is not "normal". Some of them are called flaky (ie. they sometimes failed), but most of them are valide. I often use TDD to implements new feature, and for sure I use automated spec on my local setup. Running all the specs locally is too long, and we use CI to do it. Which issues are you facing? Could you be more precise?

**Comment by jibees** [comment]
> This particular PR should be okay now. I managed to find the bad files.

There is still one conflicting file (`rubocop_todo.yml`), you probably should rebase.

Your process seems fine. I'd be happy to see which command you run for test, and which test are failing, and the associated error.

**Comment by macanudo527** [comment]
I think everything has been addressed with this one. I removed the 3 offending files.

File: app/controllers/admin/enterprise_fees_controller.rb
Comment (mkllnk) [inline app/controllers/admin/enterprise_fees_controller.rb:29]: I think that you accidentally re-added this code.

File: app/controllers/admin/enterprise_fees_controller.rb
Comment (mkllnk) [inline app/controllers/admin/enterprise_fees_controller.rb:68]: You made this better and then it got reverted again. I would simplify it like this:
```suggestion
        order_cycle ||= OrderCycle.new(coordinator: coordinator) if coordinator.present?
```

File: app/controllers/admin/enterprises_controller.rb
Comment (mkllnk) [inline app/controllers/admin/enterprises_controller.rb:183]: This also got reverted again.
```suggestion
         @order_cycle ||= OrderCycle.new(coordinator: coordinator) if coordinator.present?
```

File: app/controllers/admin/enterprises_controller.rb
Comment (mkllnk) [inline app/controllers/admin/enterprises_controller.rb:210]: Somehow this comment came back.

File: spec/system/admin/order_spec.rb
Comment (mkllnk) [inline spec/system/admin/order_spec.rb:650]: This is more than a style change. The whole spec seems rewritten. Was that intended? It would be better to explain such changes in a separate commit. But it could just be a merge conflict that wasn't resolved properly.

File: spec/system/admin/enterprise_fees_spec.rb
Comment (mkllnk) [inline spec/system/admin/enterprise_fees_spec.rb:163]: This spec also seems rewritten with several specs missing.

File: spec/system/admin/bulk_order_management_spec.rb
Comment (mkllnk) [inline spec/system/admin/bulk_order_management_spec.rb:947]: I think that this was removed intentionally because it was leading to flaky specs. You accidentally added it when resolving merge conflicts.

**Review by mkllnk — CHANGES_REQUESTED** [review]
This looks heaps better, well done. Unfortunately the pan-ultimate commit 987ea38349f6666848ff0cad71f13d65e08a1ea7 seems to revert some changes. It looks like the merge conflict resolution wasn't done properly.

Commit 987ea38349f6666848ff0cad71f13d65e08a1ea7 also has some strange reverts in style changes.

I would suggest to remove those two commits from this pull request and then open a new pull request with further changes. It will make reviewing a lot easier and makes sure that the first part is merged sooner.

**Comment by macanudo527** [comment]
> This looks heaps better, well done. Unfortunately the pan-ultimate commit [987ea38](https://github.com/openfoodfoundation/openfoodnetwork/commit/987ea38349f6666848ff0cad71f13d65e08a1ea7) seems to revert some changes. It looks like the merge conflict resolution wasn't done properly.
> 
> Commit [987ea38](https://github.com/openfoodfoundation/openfoodnetwork/commit/987ea38349f6666848ff0cad71f13d65e08a1ea7) also has some strange reverts in style changes.
> 
> I would suggest to remove those two commits from this pull request and then open a new pull request with further changes. It will make reviewing a lot easier and makes sure that the first part is merged sooner.

You mentioned the same commit twice. Was there another commit that should be pulled from this PR?

**Comment by mkllnk** [comment]
Oh sorry, I must have mixed up some tabs. It all seems to be in the same commit.

**Comment by macanudo527** [comment]
OK, I'll split it into something more manageable. Is 50 files a PR too much?

**Review by mkllnk — APPROVED** [review]
The remaining commits look great, thank you! I'll squash two commits to remove a detour before the next person reviews.

**Comment by mkllnk** [comment]
@macanudo527 I force-pushed to your branch. You need to reset to that commit if you want to change anything here.

**Comment by macanudo527** [comment]
Should I merge this? or rebase? I'm not sure of the process and I want to double-check.

**Comment by mkllnk** [comment]
You can probably leave it as is. But if you want your local branch to be exactly the same as this one then you can run:
```
git fetch origin # assuming that's your fork
git reset origin/fix_linelength # you can add `--hard` if you know what you are doing
```

But once this PR got a second review and is tested and merged, you can just rebase your other branches on master.

File: app/services/products_renderer.rb
Comment (rioug) [inline app/services/products_renderer.rb:79]: This should be :
```ruby
elsif distributor.preferred_shopfront_product_sorting_method == "by_category" &&
      distributor.preferred_shopfront_taxon_order.present?`
```

**Review by rioug — CHANGES_REQUESTED** [review]
There were a few commits which seemed to have unrelated changes, still readability changes but not related to the commit message. 
Anyway, I just spotted a small issue, if you could fix that we should be good to go !

Great work :+1:

File: app/services/products_renderer.rb
Comment (macanudo527) [reply app/services/products_renderer.rb:79]: OK, this should be patched up now.

**Review by rioug — APPROVED** [review]
Looks good now :+1:, thanks a lot !

**Comment by rioug** [comment]
Merging as it's only code style changes.

**Pull request merged**
